### DISCUSSION TITLE
Cannot backup a KV Redis database stored on Vercel?

### DISCUSSION THREAD

**Discussion by tomatac** [vercel/community#7066] [Bug Report]
### Page to Investigate

https://vercel.com/docs/storage/vercel-kv/limits

### Steps to Reproduce

Is there a way to backup a KV Redis database stored on Vercel?
I read that recommended way is to use redis-cli package but that does not work.
Ref: https://vercel.com/docs/storage/vercel-kv/limits
"Backup/Restore not supported – Use redis-cli instead"
SAVE command returns an error:
(error) ERR Command is not available: 'SAVE'. See https://vercel.com/docs/storage/vercel-kv/vercel-kv-error-codes for details
BGSAVE command returns an error:
(error) ERR Command is not available: 'BGSAVE'. See https://vercel.com/docs/storage/vercel-kv/vercel-kv-error-codes for details

Backup is critical for a database. Who would want to use a database that cannot be backed up?

**Comment by gautamj** [comment]
I'm also looking to back up a Vercel KV database, and I just saw this: https://academy.ilyakam.com/exporting-data-out-of-vercel-kv-510f27d98afe

Curious if you've already looked into that one @tomatac ? When I try it out I can report back.

**Comment by gautamj** [comment]
update: I was able to export using upstash-redis-dump like this:
```% upstash-redis-dump -db 0 -host my-host-name.upstash.io -port [PORT] -pass [PASS] -tls > /path/to/fileToWrite```

And then I was able to import to another kv store using redis-cli like this:
```% redis-cli --tls -u redis://[USER]:[PASS]@my-other-host-name.upstash.io:[PORT] --pipe < /path/to/fileToRead```

**Reply by pawlean** [reply]
@gautamj Thanks for coming back and sharing your solution with the community!

[a comment by an automated account is not shown]
